examples/tv-app/android/java/ContentAppCommandDelegate.cpp

```diff
@@ -153,6 +147,7 @@ Status ContentAppCommandDelegate::InvokeCommand(EndpointId epId, ClusterId clust
             return chip::Protocols::InteractionModel::Status::Failure;
         }
 
+        // Q: It would seem that this is the successfull return case (?)
```

Comment:
Status 0 indeed means success, but I don't think this resolves my question because there does not seem to be a path where a 0 Status is returned:

`value[FAILURE_STATUS_KEY]` is only being returned on line 145, which is only being parsed if `value[FAILURE_KEY]` is non-empty - indicating that the Status value also will be non-zero because there is an error response. The Failure status should be one of the failure responses formatted [here](https://github.com/project-chip/connectedhomeip/blob/master/examples/tv-app/android/App/platform-app/src/main/java/com/matter/tv/server/service/ContentAppAgentService.java#L151).

This code looks wrong to me because InvokeCommand returns a Status but the command response is only parsed (to read a status field) if there in the failure case. The returned Status will always be a failure case, but only 'valid' if there was an actual failure case - which the caller can then never really know. In `AccountLoginManager::GetSetupPIN` the returned Status is ignored, and the response is parsed explicitly with a call to `mCommandDelegate->FormatGetSetupPINResponse` to parse the response object. But this is only valid to do if InvokeCommand returns successfully - which it seems like it will never do. 

It seems like the Status returned from InvokeCommand on line [156](https://github.com/project-chip/connectedhomeip/blob/master/examples/tv-app/android/java/ContentAppCommandDelegate.cpp#L156) in the current main branch could/should be Success, to indicate that the command returned a non-failure response, which can then be parsed by the caller. But it returns Failure status (UnSupportedEndpoint) instead.

I'm hesistant to start changing things here before I can get confirmed if there is something I'm missing or if indeed this code is ambiguous.

Reply:
It looks to me like the only caller of this method is AccountLoginManager::GetSetupPin and it ignores the return value, so you may be right that it never returns success. As a result, its safe for you to make changes so that it returns a sane value.

Reply:
Fixed in latest commit.